src/common/hook/useRevokeQueue/useRevokeQueue.test.tsx

```diff
@@ -0,0 +1,64 @@
+import { act, renderHook } from "@testing-library/react-hooks";
+import { getDefaultProvider, Wallet } from "ethers";
+import { RevokeState } from "../../../constants/RevokeState";
+import { revokeDocumentJob } from "../../../services/revoking";
+import sampleConfig from "../../../test/fixtures/sample-config-ropsten.json";
+import sampleJobs from "../../../test/fixtures/sample-revoke-jobs.json";
+import { Config } from "../../../types";
+import { useRevokeQueue } from "./useRevokeQueue";
+import { getRevokingJobs } from "./utils/revoke";
+import sampleWrappedDocument from "../../../test/fixtures/sample-wrapped-document.json";
+
+jest.mock("../../../services/revoking");
+jest.mock("./utils/revoke");
+
+const mockRevokeJob = revokeDocumentJob as jest.Mock;
+const mockGetRevokeJobs = getRevokingJobs as jest.Mock;
+
+const config = {
+  ...sampleConfig,
+  wallet: Wallet.createRandom().connect(getDefaultProvider("ropsten")),
+} as Config;
+
+const revokeDocumentEntries = [sampleWrappedDocument];
+
+describe("useRevokeQueue", () => {
+  it("should have the correct initial state", () => {
+    const { result } = renderHook(() => useRevokeQueue(config, revokeDocumentEntries));
+    expect(result.current.revokeState).toBe(RevokeState.UNINITIALIZED);
+    expect(result.current.revokedDocuments).toStrictEqual([]);
+  });
+
+  it("should revoke correctly and correctly inserted into revokedDocuments", async () => {
+    mockGetRevokeJobs.mockReturnValueOnce(sampleJobs);
+    mockRevokeJob.mockResolvedValue("tx-id");
```

Comment:
how this works?

Reply:
cos above i mock the function, so this will resolve/return the value once the function is called. 
For mockRevokeJob it should return a transaction hash string so in the test we mock it like that.
For mockGetRevokeJobs we should get an array of jobs, so we mock the return object with sampleJobs.